Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py

```diff
@@ -0,0 +1,1653 @@
+                                'value': split_filter[1]})
+    return filters_to_send
+
+
+def handle_tags(tags: Optional[List[str]]) -> List[Dict[str, Any]]:
+    """
+    Creates the list of tags in the format that the request expects.
+    The tags are given from the user as a comma-separated list, in the format of tagkey=tagvalue.
+    """
+    tags_to_send = []
+    if tags:
+        for tag in tags:
+            split_tag = tag.split('=')
+            if len(split_tag) != 2 or not split_tag[0] or not split_tag[1]:
+                raise DemistoException('Tags should be in the format of "tagkey1=tagvalue1,tagkey2=tagvalue2". '
+                                       f'The tag "{tag}" doesn\'t meet this requirement.')
+            tags_to_send.append({'key': split_tag[0],
+                                 'value': split_tag[1]})
+    return tags_to_send
+
+
+def validate_array_arg(array_arg: List[str], arg_name: str, arg_options: List[str]) -> None:
+    """
+    Validates that all comma-separated provided arg values are in the available options.
+    """
+    if not set(array_arg).issubset(arg_options):
+        raise DemistoException(f'{arg_name} values are unexpected, must be of the following: {", ".join(arg_options)}.')
+
+
+def error_file_list_command_args_validation(source_types: List[str], categories: List[str], statuses: List[str],
+                                            file_types: List[str], search_options: List[str], severities: List[str]) -> None:
+    """
+    Validates values for all multi-select args in 'prisma-cloud-error-file-list' command.
+    """
+    validate_array_arg(categories, 'Categories', CATEGORIES_OPTIONS)
+    validate_array_arg(file_types, 'File types', FILE_TYPES_OPTIONS)
+    validate_array_arg(search_options, 'Search options', SEARCH_OPTIONS_OPTIONS)
+    validate_array_arg(severities, 'Severities', SEVERITIES_OPTIONS)
+    validate_array_arg(source_types, 'Source types', SOURCE_TYPES_OPTIONS)
+    validate_array_arg(statuses, 'Statuses', STATUSES_OPTIONS)
+
+
+def remove_empty_values_from_dict(dict_to_reduce: Dict[str, Any]):
+    """
+    Removes empty values from given dict and from the nested dicts and lists in it.
+    """
+    reduced_dict = {}
+    for key, value in dict_to_reduce.items():
+        if not value:
+            continue
+
+        if isinstance(value, dict):
+            if reduced_nested_dict := remove_empty_values_from_dict(value):
+                reduced_dict[key] = reduced_nested_dict
+
+        elif isinstance(value, list):
+            reduced_list = []
+            for item in value:
+                if isinstance(item, dict):
+                    if reduced_nested_dict := remove_empty_values_from_dict(item):
+                        reduced_list.append(reduced_nested_dict)
+                elif item:
+                    reduced_list.append(item)
+            if reduced_list:
+                reduced_dict[key] = reduced_list
```

Comment:
can we just do `reduced_dict[key] = remove_empty_values_from_dict(subvalue) for subvalue in value`?

Reply:
No, it doesn't have the same behaviour- can cause an empty list being inserted in case all nested values are empty.
I did change the function to make it more generic and recursive, and less nested.